docs/docs/tooling/profiler.md

```diff
@@ -76,60 +92,146 @@ unconstrained fn zero_out_array(ptr: u32, mut array: [u32; 32]) -> [u32; 32] {
+
+:::
+
+### Profiling proving backend gates
 
-### Generate an unconstrained execution trace flamegraph
+The profiler further provides the ability to flamegraph a Noir program's proving backend gates footprint. This is useful for fully identifying and optimizing proving bottlenecks of Noir programs.
 
-The profiler also enables developers to generate a flamegraph of the unconstrained execution trace. For unconstrained functions Noir compiles down to Brillig bytecode, thus we will be seeing a flamegraph of Brillig opcodes, rather than ACIR opcodes.
+This feature depends on the proving backend you are using. We will use [Barretenberg](https://github.com/AztecProtocol/aztec-packages/tree/master/barretenberg) as an example here. Follow the [quick start guide](../getting_started/quick_start.md#proving-backend) to install it if you have not already.
+
+#### Updating the demonstrative project
+
+Let's first revert our demonstrative program into a constrained program by removing the `unconstrained` modifier from the main function:
+
+```rust
+fn main(...){...}
+```
+
+Then recompile the program running `nargo compile`.
+
+#### Flamegraphing
+
+Let's take a granular look at our program's proving backend gates footprint using the profiler, running:
 
-Let's take our initial program and simply add an `unconstrained` modifier before main (e.g. `unconstrained fn main`). Then run the following command:
 ```sh
-noir-profiler execution-opcodes --artifact-name ./target/program.json --prover_toml_path Prover.toml --output ./target
+noir-profiler gates --artifact-path ./target/program.json --backend-path bb --output ./target
 ```
-This matches the `opcodes` command, except that now we need to accept a `Prover.toml` file to profile execution with a specific set of inputs.
 
-We will get the following flamegraph with 1,582 opcodes executed:
-![Brillig Trace Initial Program](@site/static/img/tooling/profiler/brillig-trace-initial-32.png)
+The `--backend-path` flag takes in the path to your proving backend binary.
 
-Circuit programming (ACIR) is an entirely different execution paradigm compared to regular programming. To demonstrate this point further, let's generate an execution trace for our optimized ACIR program once we have modified `main` to be `unconstrained`.
+The above command assumes you have Barretenberg (bb) installed and that its path is saved in your PATH. If that is not the case, you can pass in the absolute path to your proving backend binary instead.
 
-We then get the following flamegraph with 2,125 opcodes executed:
-![Brillig Trace "Optimized"](@site/static/img/tooling/profiler/brillig-trace-opt-32.png)
+Flamegraph of the demonstrative project generated with bb v0.76.4:
+
+![Gates Flamegraph Optimized](@site/static/img/tooling/profiler/gates-flamegraph-optimized.png)
```

Comment:
Should we show the unoptimized one before this?

Reply:
Bumped the optimized flamegraph as this reads after the optimizing section.

Happy to reposition according to any reordering of sections from the parallel discussion above.